crypto/asn1/asn_mime.c

```diff
@@ -300,6 +300,8 @@ int SMIME_write_ASN1_ex(BIO *bio, ASN1_VALUE *val, BIO *data, int flags,
 
     if (ctype_nid == NID_pkcs7_enveloped) {
         msg_type = "enveloped-data";
+    } else if (ctype_nid == NID_id_smime_ct_authEnvelopedData) {
+        msg_type = "authEnveloped-data";
```

Comment:
I assume this name is coming from here?
https://datatracker.ietf.org/doc/html/rfc8551#section-3.2.2

Reply:
Yes exactly, it's from S/MIME 4.0 spec.

Reply:
Btw. I found it when doing integration testing with BouncyCastle (the test examples can be seen here: https://github.com/bukka/jcrypto/tree/1028865c22972b1af358dc020222ae4bc57fabd7/examples/mail ). In this case BouncyCastle produces the right type but OpenSSL produces nothing. There are actually more differences and issues (mostly in OpenSSL) which I plan to look into it later.

Reply:
Actually this comes from changes that are not in the BouncyCastle yet but created a PR for them: https://github.com/bcgit/bc-java/pull/1791 and also explicitly added to my app. So it's not exactly incompatibality with BouncyCastle yet.. :) But the other issues are already incompatible in latest versions (e.g. KEK is different structure and some other things).